api/src/main/java/org/jboss/forge/roaster/model/util/Refactory.java

```diff
@@ -391,4 +393,45 @@ public static void createToStringFromFields(final JavaClassSource clazz,
                "return result;";
       method.setBody(body);
    }
+
+   /**
+    * Replaces existing implementations of equals(Object) and hashCode()
+    * using all final fields of the specified {@link JavaClassSource}.
+    * If a method does not exist, it will not be created. If there are
+    * no more fields available, existing methods will be deleted without
+    * replacement.
+    *
+    * @param source the source to generate equals(Object) and hashCode() for, must not be {@code null}.
+    */
+   public static void regenerateEqualsAndHashCode(final JavaClassSource source)
+   {
+      List<FieldSource<JavaClassSource>> finalFieldsToUse = new LinkedList<FieldSource<JavaClassSource>>();
+      for (PropertySource<JavaClassSource> property : source.getProperties())
+      {
+         if (property.getField() != null && property.getField().isFinal())
```

Comment:
This doesn't check static fields, but I think we shouldn't assume anything. I like the idea of introducing field selectors or just passing the list of fields should be enough(the caller can use the streams API to filter accordingly)